test/verify/check-login

```diff
@@ -169,6 +169,44 @@ account    required     pam_succeed_if.so user ingroup %s""" % m.get_admin_group
 
         self.allow_restart_journal_messages()
 
+
+    @skipImage("logs in via ssh, not cockpit-session", "fedora-coreos")
+    def testLogging(self):
+        m = self.machine
+        b = self.browser
+
+        # Clean out the relevant logfiles
+        m.execute("truncate -s0 /var/log/{[bw]tmp,lastlog} /var/run/utmp")
+
+        m.start_cockpit()
+        b.login_and_go('/system')
+        time.sleep(2)
+        assert not b.is_present('#last-login')
+        assert not b.is_present('#last-failed')
```

Comment:
Nack, no sleeps in tests. `b.wait_not_present('#last-login')`

Reply:
This is no good.  I'm not waiting for it not to be present.  I'm waiting to make sure it doesn't become present.  Is there a better way to say "wait until the page is fully loaded and there are no more outstanding requests on the cockpit websocket?"

Reply:
This wouldn't be a well-defined state, as there are pretty much always websocket requests, due to having signal listeners. In most cases this can be done by rearranging the test to *first* make sure that the thing appears, and *afterwards* changing the system state so that it will disappear,  and using `wait_not_present()`. This might not work here, as I suppose you can't remove the fact that the previous login failed while the session is running.

The next best approach is to make the code change some `data-stable=yes` attribute, so that the test suite can differ between "stuff is still loading" and "stuff has run with a negative result". `git grep data-stable` shows some precedent there.